### PR TITLE
Update contribution instructions

### PR DESCRIPTION
**Pull request by astrojuanlu**
This updates the contribution instructions, and avoids having them in two places by sourcing the Sphinx docs from the root directory.

It is still a draft because I started by making some modifications without changing the structure much, but I would like to improve the document so it's clear that one can contribute without setting up a local development environment, just using the GitHub UI.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Rendered version: https://poliastro--1321.org.readthedocs.build/en/1321/contributing.html

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,118 +1,182 @@
...
-Not only things break all the time, but also different people have
-different use cases for the project. If you find anything that doesn\'t
-work as expected or have suggestions, please refer to the [issue
-tracker](https://github.com/poliastro/poliastro/issues) on GitHub.
+### Report bugs
 
-## Documentation
+Not only things can break, but also different people have
+different use cases for the project. If you find anything that doesn't
+work as expected or have suggestions, open a new issue on our
```
Comment (Yash-10) [inline CONTRIBUTING.md:11]: What do you think of this?

```suggestion
work as expected or have suggestions, please open a new issue on our
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,118 +1,182 @@
...
+Once you have finished it, you can edit the poliastro source files
+straight from GitHub and open pull requests from your browser.
 
-You will also need to understand how git works. git is a decentralized
+### Work locally
+
+For more complex contributions, it will be more effective
+to set up a local development environment and work from your own computer.
+
+The most important thing is to understand how git works. git is a decentralized
```
Comment (Yash-10) [inline CONTRIBUTING.md:57]: Does the "g" in "git" generally written uppercase, i.e. Git instead of git?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,118 +1,182 @@
...
-## Development environment
-
-These are some succint steps to set up a development environment:
+at the same time. To learn more, you can have a look at
+[this Getting Started guide](https://docs.github.com/en/get-started/getting-started-with-git).
+
+Finally, you might want to have an Integrated Development Environment (IDE)
+that makes editing the source files easier.
+If you are hesitant on what IDE or editor to use,
+have a look at PyCharm, Visual Studio Code, or Spyder.
```
Comment (Yash-10) [inline CONTRIBUTING.md:66]: Would adding the official web links for the IDEs described here, be a good choice?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,118 +1,182 @@
...
+9. Upgrade the development dependencies using `python pip install -U pip setuptools wheel flit tox`
+10. Install the code in development mode using `flit install --simlink`
+    (this means that the installed code will change as soon as you change it in the
+    download location).
+
+And with this, you will be ready to start contributing!
+
+### Pull request workflow
+
+Every time you want to contribute some code or documentation to poliastro
```
Comment (Yash-10) [inline CONTRIBUTING.md:118]:
```suggestion
Every time you want to contribute some code or documentation to poliastro,
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,118 +1,182 @@
...
+you will need follow these steps:
```
Comment (Yash-10) [inline CONTRIBUTING.md:119]:
```suggestion
you will need to follow these steps:
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,118 +1,182 @@
...
+
+Remember that, whenever you want to start a new pull request,
+you need to start from step 1.
+
+### Build the documentation
+
+To build the docs, run:
+
+```console
+(.venv) $ cd docs
```
Comment (Yash-10) [inline CONTRIBUTING.md:147]: Was the inclusion of `(.venv)` intentional?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,118 +1,182 @@
...
+   to get inspiration from. All new features and fixes should be tested,
+   and in the ideal case the coverage rate should increase or stay the same.
+2. To check if your code is correct, run `tox -e reformat,check,py39-fast`
+   (`py39` stands for Python 3.9, you can replace it by `py38`).
+3. Notice that you can run a subset of the tests by
+   passing extra arguments to pytest, for example running
+   `tox -e py39-fast -- -k "anomaly"`
+
+Automatic services will ensure your code works
+on all the supported operative systems and Python versions.
```
Comment (Yash-10) [inline CONTRIBUTING.md:182]: Maybe you meant "operating"? :)
```suggestion
on all the supported operating systems and Python versions.
```

**Review by Yash-10 — COMMENTED** [review]
Tried to give a review!

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,118 +1,182 @@
...
+5. Add an `upstream` remote with `git remote add upstream https://github.com/poliastro/poliastro.git`
+   and fetch its information with `git fetch upstream`
+6. Set your `main` branch to track `upstream` using `git branch --set-upstream-to=upstream/main`
+
+Next, configure your Python environment:
+
+6. Install Python for development.
+7. Create a Python virtual environment using `python -m venv .venv`
+8. Activate it using `source .venv/bin/activate`
+9. Upgrade the development dependencies using `python pip install -U pip setuptools wheel flit tox`
```
Comment (jorgepiloto) [inline CONTRIBUTING.md:109]: Shouldn't this line be `python -m pip...`? 🤔

**Review by jorgepiloto — COMMENTED** [review]
Just left one comment @astrojuanlu, as @Yash-10 already made a deep review about this 🚀

File: CONTRIBUTING.md (hunk 1 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:11]: Good! I removed excessive use of "please", but in this case it makes sense

File: CONTRIBUTING.md (hunk 2 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:57]: You're right, in the official site they call it `Git` https://git-scm.com/ Will change it

File: CONTRIBUTING.md (hunk 3 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:66]: Good idea, I was a bit lazy 😂 will do!

File: CONTRIBUTING.md (hunk 8 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:109]: In a non-Windows environment it's not that important, and also the virtual environment is activated at this point, but you're right that the core devs are trying to push for it more, and it's only a bit of extra typing. Will change it.

File: CONTRIBUTING.md (hunk 6 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:147]: Yes! It's because the virtual environment is activated

File: CONTRIBUTING.md (hunk 7 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:182]: I always make this mistake 🙈

File: CONTRIBUTING.md (hunk 8 above)
Comment (astrojuanlu) [reply CONTRIBUTING.md:109]: Hah, I see what you meant - I forgot the `-m` 🙈

**Comment by astrojuanlu** [comment]
Thanks @jorgepiloto and @Yash-10 😍 I addressed all your comments

**Review by Yash-10 — APPROVED** [review]
Looks great! :smiley:

**Comment by astrojuanlu** [comment]
Merging! 🚀

**Pull request merged**